### DISCUSSION TITLE
Using multiple location managers with new dependency style

### DISCUSSION THREAD

**Discussion by mbrandonw** [pointfreeco/composable-core-location#16] [General]
There was a question posted on a comment of PR #12 and I thought it'd be better to address in a discussion:

https://github.com/pointfreeco/composable-core-location/commit/92d4cad059e369f2e5449fd632bbe6fc8f99b182#commitcomment-56249518

> @stephencelis quick question, how would one use then multiple instances of LocationManager without ids? I'm using at least 2 in parallel with different accuracy and other settings.

@andreyz This is a good question, and we should update the docs to explain more.

The main reason we made this change is because we feel that the most common usage of location managers is to just have a single one, and the old API was designed to be able to support any number of managers. That API was kind of complicated since you had to juggle ids. So it didn't make sense to us to force people who just want a single manager to have to think about ids.

Having said that, if you do have a use case for multiple managers, then you will simply hold multiple managers in your environment:

```swift
struct AppEnvironment {
  var accurateLocationManager = LocationManager.live
  var inaccurateLocationManager = LocationManager.live
}
```

Both of those managers are completely independent and powered by distinct `CLLocationManager`s under the hood.

**Comment by andreyz** [comment]
@mbrandonw thank you so much for quick and helpful response!
